recommender-back/src/apis/current.py

```diff
@@ -64,6 +89,13 @@ def find_nearest_stations_weather_data(self, poi: PointOfInterest):
                     if key in missing_fields:
                         missing_fields.remove(key)
             if not missing_fields or not weather:
+                smallest, nearest = float('inf'), ''
+                for station in aqi:
+                    dist = abs(aqi[station]['Latitude'] - lat)\
+                                            + abs(aqi[station]['Longitude'] - lon)
+                    if dist < smallest:
+                        smallest, nearest = dist, station
+                returned.setdefault('Air quality', aqi[nearest]['Air quality'])
```

Comment:
Maybe consider not adding aqi as a POI key, and instead having aqi stations' current and forecast values be its own api? This would make integrating to Mongo pretty straightforward I think :thinking:

Reply:
Could not reproduce the 500 response, but added a check that if all AQI stations have nan values. AQI stations only have the current data, the forecast AQI will have it's own implementation.

Reply:
Fixed the issue, sometimes all AQI stations return nan values.